app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/ApplicationController.scala

```diff
@@ -50,17 +52,50 @@ class ApplicationController @Inject() (
 
   def startSubscription(service: Service): Action[AnyContent] = ggAuthorisedUserWithEnrolmentsAction {
     implicit request => implicit loggedInUser: LoggedInUserWithEnrolments =>
-      enrolledForService(loggedInUser, service) match {
-        case Some(_) =>
+      try {
+        if (isUserEnrolledFor(loggedInUser, service)) throw SpecificEnrolmentExists(service)
+
+        loggedInUser.groupId match {
+          case Some(groupId) =>
+            hasGroupIdEnrolmentTo(groupId, service).flatMap { groupIdEnrolmentExists =>
+              if (groupIdEnrolmentExists) throw SpecificGroupIdEnrolmentExists(service)
+
+              cdsEnrolmentCheck(loggedInUser, groupId, service)
+            }
+          case None if enrolledCds(loggedInUser).isDefined =>
```

Comment:
Would "if (isUserEnrolledFor(loggedInUser, CDS))" be clearer here?

Reply:
yes, I missed that place when create `isUserEnrolledFor` method.

Thanks